server/src/main/java/org/elasticsearch/monitor/fs/FsHealthService.java

```diff
@@ -150,7 +153,17 @@ public void run() {
 
         private void monitorFSHealth() {
             Set<Path> currentUnhealthyPaths = null;
-            for (Path path : nodeEnv.nodeDataPaths()) {
+            brokenLock = false;
```

Comment:
Clearing this flag here may mean the node reports itself as healthy even though it hasn't actually passed this health check yet. I think we should clear this flag only after setting `unhealthyPaths` at the very bottom of this method.

Reply:
sure, done.